File: experiments/runs/2026-09-20_07-18-13_claim-office-java-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/main/java/PriceList.java
import java.math.BigDecimal;
import java.util.Map;

/** The MHPCO's venerable price list of insurance values and base premiums. */
public final class PriceList {

    private static final Map<String, BigDecimal> MAIN_ITEMS = Map.of(
            "sword", BigDecimal.valueOf(1000),
            "amulet", BigDecimal.valueOf(600),
            "staff", BigDecimal.valueOf(800),
            "potion", BigDecimal.valueOf(400));

    private static final BigDecimal COMPONENT_VALUE = BigDecimal.valueOf(250);

    private static final Map<String, BigDecimal> COMPONENTS = Map.of(
            "rune", COMPONENT_VALUE,
            "moonstone", COMPONENT_VALUE);

    /** Base premium is always a tenth of the insurance value. */
    private static final BigDecimal PREMIUM_RATE = BigDecimal.valueOf(10);

    private PriceList() {
    }

    public static boolean isComponent(String type) {
        return COMPONENTS.containsKey(type);
    }

    public static boolean isKnown(String type) {
        return MAIN_ITEMS.containsKey(type) || COMPONENTS.containsKey(type);
    }

    public static BigDecimal insuranceValue(String type) {
        BigDecimal value = MAIN_ITEMS.get(type);
        if (value == null) {
            value = COMPONENTS.get(type);
        }
        if (value == null) {
            throw new ScenarioException("unknown item type: " + type);
        }
        return value;
    }

    public static BigDecimal basePremium(String type) {
        return insuranceValue(type).divide(PREMIUM_RATE);
    }
}

File: experiments/runs/2026-09-23_07-14-12_claim-office-example-mapping_exact-ptdd-v1-cc_opus-5-5-no-thinking/src/priceList.ts
export interface Item {
  type: string;
  material?: string;
  enchantment?: number;
  cursed?: boolean;
}

interface PriceListEntry {
  insuranceValue: number;
  basePremium: number;
}

const MAIN_ITEMS: Record<string, PriceListEntry> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};
const COMPONENT: PriceListEntry = { insuranceValue: 250, basePremium: 25 };
const COMPONENT_TYPES = ["rune", "moonstone"];

export function isComponent(type: string): boolean {
  return COMPONENT_TYPES.includes(type);
}

function priceListEntry(type: string): PriceListEntry {
  if (isComponent(type)) return COMPONENT;
  if (!Object.hasOwn(MAIN_ITEMS, type)) throw new Error(`Unknown item type: ${type}`);
  return MAIN_ITEMS[type];
}

export function basePremium(type: string): number {
  return priceListEntry(type).basePremium;
}

export function insuranceValue(type: string): number {
  return priceListEntry(type).insuranceValue;
}
